src/dotnet/APIView/APIView/CodeFileRenderer.cs

```diff
@@ -80,6 +130,55 @@ private void Render(List<CodeLine> list, IEnumerable<CodeFileToken> node, bool s
             }
         }
 
+        private string SanitizeLineClass(string lineClass)
+        {
+            if (!String.IsNullOrEmpty(lineClass))
+            {
+                var result = lineClass.ToLower();
+                result = Regex.Replace(result, "[^a-z_0-9-]", "");
+                result = Regex.Replace(result, "^[0-9]+", "");
+                result = Regex.Replace(result, "//s+", "");
+                return result;
+            }
+            return lineClass;
+        }
+
+        private string SanitizeLineId(string lineId, HashSet<string> lineIds)
+        {
+            int resultAsInt;
+            if (Int32.TryParse(lineId, out resultAsInt))
+            {
+                return resultAsInt.ToString();
+            }
+
+            if (!String.IsNullOrWhiteSpace(lineId))
+            {
+                var result = lineId.ToLower();
+                result = Regex.Replace(result, "[^a-z_0-9-:.]", "");
+                result = Regex.Replace(result, "^[0-9]+", "");
+                result = Regex.Replace(result, "//s+", "");
+
+                if (lineIds.Contains(result))
+                {
+                    do
```

Comment:
would it be good to add a comment with an example of this? it could be my lack of context on the project but all those `1` seem like magic :|